k8s/monitoring/kubemon.py

```diff
@@ -0,0 +1,100 @@
+"""
+Monitor Global.health Kubernetes pods to see if any
+are stuck in Pending status for a while. If such pods are
+present, notify on Slack.
+
+IMPORTANT:
+  The kubernetes Python package needs to be kept in sync
+  with the EKS Kubernetes version. The compatibility matrix
+  for the client Python library is at
+  https://github.com/kubernetes-client/python#compatibility
+
+NOTE:
+  This will need to be updated once we move to separate
+  clusters for different environments.
+"""
+
+import os
+import sys
+import base64
+import logging
+import datetime
+from typing import Any
+from pathlib import Path
+
+import requests
+import kubernetes
+
+def setup_logger():
+    h = logging.StreamHandler(sys.stdout)
+    rootLogger = logging.getLogger()
+    rootLogger.addHandler(h)
+    rootLogger.setLevel(logging.INFO)
+
+
+SERVICES = ["curator", "data", "location"]
+SLACK_WEBHOOK_URL = os.getenv("SLACK_WEBHOOK_URL")
+
+if (
+    kubeconfig_path := Path.home() / ".kube" / "config"
+).exists() and kubeconfig_path.stat().st_size > 0:
+    logging.info("Using existing ~/.kube/config")
+elif KUBECONFIG_VALUE := os.getenv("KUBECONFIG_VALUE"):
+    logging.info("Writing base64 encoded KUBECONFIG_VALUE to ~/.kube/config")
+    (Path.home() / ".kube").mkdir(exist_ok=True)
+    logging.info(kubeconfig_path)
+    kubeconfig_path.write_text(base64.b64decode(KUBECONFIG_VALUE).decode("utf-8"))
+elif os.getenv("ENV") == "test":
+    logging.info("Skipping kubernetes configuration in testing")
+else:
+    logging.error("Neither ~/.kube/config or KUBECONFIG_VALUE exists, aborting")
+    sys.exit(1)
+
+
+def last_status(pod: dict[str, Any]) -> (str, datetime.datetime):
+    "Returns last status for pod"
+    statuses = list(pod.status.conditions)
+    statuses.sort(key=(lambda k: k.last_transition_time), reverse=True)
+    return statuses[0].type, statuses[0].last_transition_time
+
+
+def is_service(pod: dict[str, Any]) -> bool:
+    "Returns whether pod is curator, location, or data pod"
+    return any(pod.metadata.name.startswith(service + "-") for service in SERVICES)
+
+
+def get_pods() -> list[dict[str, Any]]:
+    "Returns list of pods in the cluster"
+    v1 = kubernetes.client.CoreV1Api()
+    return filter(is_service, v1.list_pod_for_all_namespaces(watch=False).items)
+
+
+def is_pending(pod: dict[str, Any]) -> bool:
+    "Returns whether a pod is pending"
+    return last_status(pod)[0] != "Ready"
```

Comment:
`== "Pending"`? or alert on not being ready?

Reply:
changed the method to is_not_ready